js/prefs/switchencryptionoptions.js

```diff
@@ -0,0 +1,97 @@
+            this.showKeys();
+        }
+
+    },
+
+    // shows the keys of the identity
+    showKeys: function () {
+
+        // get the identityID
+        let identity = $("identity");
+        let identityID = Number($F(identity));
+
+        HordeCore.doAction('getIdentityPubKey', // NOTE: methods from doAction cannot get any parameters! They expect $this->vars->$name
+            {
+                identityID: identityID,
+            },
+            {
+                callback: function (response) {
+                    // check for keys in the adressbook for the identity and show them
+
+                    let div = $$('div.prefsSmimeContainer')[0];
+
+                    // ajax reloads things for three times for some reason, so make sure that nothing appears additionally. Remove extra appearances.
+                    if (div.previous('ul#addressbookpubkey')){
+                        $('addressbookpubkey').remove();
+                        $('addressbookonlyinfos').remove();
+                    }
+
+                    if ($('smimeselect').checked === true){
+                        $('addressbookpubkey').remove();
+                        $('addressbookonlyinfos').remove();
+                    }
+                    
+                    // if smime is not selected, some information on the keys in the adressbook shows
+                    if ($('smimeselect').checked === false) {
+
+                        let ul = new Element('ul', {
+                            id: "addressbookpubkey"
+                        });
+
+                        for (let key in response) {
+                            if (response.hasOwnProperty(key)) {
+
+                                let li = new Element('li').setStyle({'list-style-type':'none'});
+                                let a = new Element('a', {
+                                    href: response[key]
+                                })
+                                a.update(key);
+                                li.insert(a);
+                                ul.insert(li);
+                            }
+                        }
+
+                        div.insert({before: ul});
+
+                        // create a text to inform the user
+                        let text = "The following key from the addressbook is used for this identity if you want to use the addressbook only (without SMIME-keys):";
+                        
+                        let infodiv = new Element('div', {
+                            id: "addressbookonlyinfos"
+                        });
+                        infodiv.update(text);
+                        ul.insert({before: infodiv});
+                    }
+                }
+            }
+        );
+
+    }
+};
+
+// on change of identity show chosen options for SMIME manament or Adressbook keys management
+document.observe('HordeIdentitySelect:change', switchEncryption.checkSMIMEOption.bindAsEventListener(switchEncryption));
```

Comment:
Use document.addEventListener('eventname', function(event) { } for document.observe replacements.

Reply:
ok did that thanks!